docs-source/docs/modules/cqrs/examples/shopping-cart-service-scala/src/test/scala/sample/shoppingcart/IntegrationSpec.scala

```diff
@@ -138,31 +157,33 @@ class IntegrationSpec
       }
     }
 
-    "update and consume from different nodes" in {
-      val cart1 = ClusterSharding(testKit1.system).entityRefFor(ShoppingCart.EntityKey, "cart-1")
-      val probe1 = testKit1.createTestProbe[StatusReply[ShoppingCart.Summary]]
-
-      val cart2 = ClusterSharding(testKit2.system).entityRefFor(ShoppingCart.EntityKey, "cart-2")
-      val probe2 = testKit2.createTestProbe[StatusReply[ShoppingCart.Summary]]
-
+    "update and consume from different nodes via gRPC" in {
       val eventProbe3 = testKit3.createTestProbe[ShoppingCart.Event]()
       testKit3.system.eventStream ! EventStream.Subscribe(eventProbe3.ref)
 
       // update from node1, consume event from node3
-      cart1 ! ShoppingCart.AddItem("foo", 42, probe1.ref)
-      probe1.receiveMessage().isSuccess should ===(true)
+      val response1 = client1.addItem(proto.AddItemRequest(cartId = "cart-1", itemId = "foo", quantity = 42))
+      val updatedCart1 = response1.futureValue.cart.get
+      updatedCart1.items.head.itemId should ===("foo")
+      updatedCart1.items.head.quantity should ===(42)
       eventProbe3.expectMessage(ShoppingCart.ItemAdded("cart-1", "foo", 42))
 
       // update from node2, consume event from node3
```

Comment:
```suggestion
      // add from client2, expect event on node3
```
Maybe move the asserting up here.